src/cycax/cycad/assembly.py

```diff
@@ -53,13 +54,31 @@ def render(self):
         logging.info("moving to the assembler")
         self.assembler.assembly_openscad()
 
-    def save(self):
+    def save(self, path: Path | None = None):
         """
-        !!!!There is a comment about this method in the jira ticket to assert its relevance.
+        Save the assembly and added part to JSON files.
+        Args:
+            path: The location where the assembly is stored. A directory for each part will be created in this path.
         """
+
+        if path is None:
+            path = Path(".")
+        if not path.exists():
+            msg = f"The directory {path} does not exists."
+            raise FileNotFoundError(msg)
+
         for item in self.pieces:
-            self.add(item)
-        self.export()
+            item.save(path)
+
+        data = self.export()
+        data_filename = path / "assembly.json"
```

Comment:
Assembly becomes a part, a complex part. So I think it should be saved as ./box/box.json if it is box. rather than ./assembly.json because that is just making unecessary mess. I know that my add method had flaws but I prefer it to this. How are you going to manage if there are 2 sides that both need the same features. A bad method is better than writing twice as many .subtract() functions. I would rather be tasked with fixing the add method than it just disappearing.

Reply:
### Directory structure
Mess vs. confusion. When Cycad is used in an assembly and the assembly files go into a directory just like parts it creates a very weird file hierarchy. Both parts and the assembly are nested at the same level as the source. The user is building and assembly so give them the assembly files right there. Next to the source, like LaTeX. For parts only the same argument would have worked but there I prefer consistency so part is always bundled into a directory.
This matches the current workflow for CAD where the source and assembly artifacts are in the Product directory with a directory inside for each part.

A path can be given to save for when one needs to separate source from build artifacts.

Reply:
I have mulitple assemblies in my playpen. I just see this not putting assembly in a floder resulting in a lot of mess. But I do agree with the logic. I think maybe then the whole part should be written into an assembly file. So if you are assembling box you would have ./box/part/part.json and ./box/assembly.json. This would cause a lot of issue though because the engineOpenSCAD does not know about the assmebly and does not need to at any poin in the proceedings.

Reply:
### Add method
I do not know what influence changes on add has with the subtract method, I'll go and have a look.
There are 2 main, and 1 behavioral  reasons for the change:

1. I changed a part and nothing worked since the change resulted in a smaller JSON file. Removing the compare fixed things. 
2. Allowing to add a part and then continue to modify it makes the code more testable. That part of add is taken over by save, so a same-same just another name.
3. add != save. Its unexpected that add creates files. We should not need the JSON file for cycad to work so allowing user to control them is fine. As long as save is called before render(). Having add so light weight is great, we will add functionality in there to allow parts lookups.

You do not have to protect a user from them selfs. Make the interface intuitive, easy to use and difficult to make mistakes but not magical (what you did there is cool). I did not expect add or save to merge my parts. If we need merge then we add it as a separate method.

I also pushed responsibility for parts save into parts.